### PR TITLE
add a few comments for the toolstate stuff

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: src/ci/docker/x86_64-gnu-tools/checktools.sh [hunk 1]
```diff
@@ -85,7 +87,8 @@ status_check() {
     check_dispatch $1 beta rls src/tools/rls
     check_dispatch $1 beta rustfmt src/tools/rustfmt
     check_dispatch $1 beta clippy-driver src/tools/clippy
-    # these tools are not required for beta to successfully branch
+    # These tools are not required on the beta/stable branches.
+    # They will still cause failure during the beta cutoff week, see `src/tools/publish_toolstate.py` for that.
     check_dispatch $1 nightly miri src/tools/miri
     check_dispatch $1 nightly embedded-book src/doc/embedded-book
     check_dispatch $1 nightly rustc-guide src/doc/rustc-guide
```
Comment (RalfJung) [inline src/ci/docker/x86_64-gnu-tools/checktools.sh:94]: @kennytm should we say that the list of "nightly" tools here and of `REGRESSION_OK` tools in `publish_toolstate.py` should be in sync?

(Ideally they could source from the same data but I am not sure what a good way to do that would be.)

Comment (kennytm) [reply src/ci/docker/x86_64-gnu-tools/checktools.sh:94]: Yes ideally these should be in sync.

File: src/ci/docker/x86_64-gnu-tools/repo.sh [hunk 2]
```diff
@@ -62,6 +62,10 @@ commit_toolstate_change() {
     MESSAGE_FILE="$1"
     shift
     for RETRY_COUNT in 1 2 3 4 5; do
+        # Call the callback; this will in the end call `change_toolstate` from
+        # `checktools.sh` if we are in the `auto` branch (pre-landing) or
+        # `src/tools/publish_toolstate.py` if we are in the `master` branch
+        # (post-landing).
```
Comment (RalfJung) [inline src/ci/docker/x86_64-gnu-tools/repo.sh:68]: @kennytm is this correct? I am not entirely sure about this.

The call with `publish_toolstate.py` is from somewhere in the CI config, which took me *forever* to find.

Comment (kennytm) [reply src/ci/docker/x86_64-gnu-tools/repo.sh:68]: Yes.

(The call from CI is `.azure-pipelines/master.yml`, you could `git grep /repo.sh` to find this. Though the comment above still refers to Travis 😓)

**Review by mark-i-m — COMMENTED** [review]
Thanks!

**Pull request merged**